cmd/terramate/cli/cloud.go

```diff
@@ -404,69 +408,49 @@ func (c *cli) detectCloudMetadata() {
 
 		for _, pull := range pulls {
 			logger.Debug().
-				Str("pull_request_url", pull.HTMLURL).
+				Str("pull_request_url", pull.GetHTMLURL()).
 				Msg("found pull request")
 		}
 
 		pull := pulls[0]
 
-		setGithubPRMetadata(md, &pull)
-
 		logger.Debug().
-			Str("pull_request_url", pull.HTMLURL).
+			Str("pull_request_url", pull.GetHTMLURL()).
 			Msg("using pull request url")
 
-		reviewRequest := &cloud.DeploymentReviewRequest{
-			Platform:    "github",
-			Repository:  c.prj.prettyRepo(),
-			URL:         pull.HTMLURL,
-			Number:      pull.Number,
-			Title:       pull.Title,
-			Description: pull.Body,
-			CommitSHA:   pull.Head.SHA,
-		}
+		setGithubPRMetadata(md, pull)
 
-		for _, l := range pull.Labels {
-			reviewRequest.Labels = append(reviewRequest.Labels, cloud.Label{
-				Name:        l.Name,
-				Color:       l.Color,
-				Description: l.Description,
-			})
+		reviews, err := listGithubPullReviews(githubClient, r.Owner, r.Name, pull.GetNumber())
+		if err != nil {
+			logger.Warn().
+				Err(err).
+				Msg("failed to retrieve PR reviews")
 		}
 
-		reviews, err := getGithubPullReviews(&ghClient, ghRepo, pull.Number)
+		checks, err := listGithubChecks(githubClient, r.Owner, r.Name, headCommit)
 		if err != nil {
 			logger.Warn().
 				Err(err).
 				Msg("failed to retrieve PR reviews")
 		}
 
-		uniqueReviewers := make(map[string]struct{})
-
-		for _, review := range reviews {
-			login := review.User.Login
-
-			if _, found := uniqueReviewers[login]; found {
-				continue
-			}
-			uniqueReviewers[login] = struct{}{}
-
-			reviewRequest.Reviewers = append(reviewRequest.Reviewers, cloud.Reviewer{
-				Login:     login,
-				AvatarURL: review.User.AvatarURL,
-			})
+		merged, err := isGithubPRMerged(githubClient, r.Owner, r.Name, pull.GetNumber())
```

Comment:
> if this instance of terramate run is running in the GHA, then querying their own "checks state" will always return 1 with the in_progress state, right?
Yes, that's actually something I was wondering myself. The checks are specific to a commit, so for my tests I was always doing `terramate run` on commits that already had their checks completed, but in case of GHA, the checks will probably still be in progress. In this case, the only solution would be that there's another mechanism to sync them later.

Reply:
Resolving this for now, we have to see how it works with some deployment runs in GHA.